source/SIL.AppBuilder.Portal/common/public/workflow.ts

```diff
@@ -0,0 +1,254 @@
+import type {
+  AnyEventObject,
+  StateMachineDefinition,
+  StateMachine,
+  TransitionDefinition,
+  TransitionDefinitionMap,
+  StateNode as XStateNode
+} from 'xstate';
+import type { RoleId } from './prisma.js';
+
+export enum ActionType {
+  /** Automated Action */
+  Auto = 0,
+  /** User-initiated Action */
+  User
+}
+
+export enum AdminLevel {
+  /** NoAdmin/OwnerAdmin */
+  None = 0,
+  /** LowAdmin */
+  Low,
+  /** Approval required */
+  High
+}
+
+export enum ProductType {
+  Android_GooglePlay = 0,
+  Android_S3,
+  AssetPackage,
+  Web
+}
+
+export type StateName =
+  | 'Readiness Check'
+  | 'Approval'
+  | 'Approval Pending'
+  | 'Terminated'
+  | 'App Builder Configuration'
+  | 'Author Configuration'
+  | 'Synchronize Data'
+  | 'Author Download'
+  | 'Author Upload'
+  | 'Product Build'
+  | 'App Store Preview'
+  | 'Create App Store Entry'
+  | 'Verify and Publish'
+  | 'Product Publish'
+  | 'Make It Live'
+  | 'Published'
+  | 'Product Creation';
+
+export type WorkflowContext = {
+  instructions:
+    | 'asset_package_verify_and_publish'
+    | 'app_configuration'
+    | 'create_app_entry'
+    | 'authors_download'
+    | 'googleplay_configuration'
+    | 'googleplay_verify_and_publish'
+    | 'make_it_live'
+    | 'approval_pending'
+    | 'readiness_check'
+    | 'synchronize_data'
+    | 'authors_upload'
+    | 'verify_and_publish'
+    | 'waiting'
+    | 'web_verify'
+    | null;
+  includeFields: (
+    | 'ownerName'
+    | 'ownerEmail'
+    | 'storeDescription'
+    | 'listingLanguageCode'
+    | 'projectURL'
+    | 'productDescription'
+    | 'appType'
+    | 'projectLanguageCode'
+  )[];
+  includeReviewers: boolean;
+  includeArtifacts: 'apk' | 'aab' | boolean;
+  start?: StateName;
+  productId: string;
+  adminLevel: AdminLevel;
+  environment: BuildEnv;
+  productType: ProductType;
+  currentState?: StateName;
+};
+
+// These are all specific to the Google Play workflows
+// Not sure how these are used, but will figure out when integrating into backend
+export type BuildEnv = {
+  googlePlayDraft?: boolean;
+  googlePlayExisting?: boolean;
+  googlePlayUploaded?: boolean;
+};
+
+export type WorkflowInput = {
+  productId?: string;
+  adminLevel?: AdminLevel;
+  environment?: BuildEnv;
+  productType?: ProductType;
+};
+
+/** Used for filtering based on AdminLevel and/or ProductType */
+export type MetaFilter = {
+  level?: AdminLevel | AdminLevel[];
+  product?: ProductType | ProductType[];
```

Comment:
I did it this way, because in most cases I am only filtering on one product type or admin level. There are a *few* instances where it was necessary to filter on multiple types. I mostly just didn't want to create unnecessary arrays when there's only one element to filter on.

Reply:
Actually, I will go ahead and fix that everywhere that I did that in my PR, not just the meta filtering